### ISSUE TITLE
Cannot set a custom location for path based tables

### ISSUE THREAD

**Issue by pietro-agilelab** [apache/iceberg#8377]
### Query engine

PySpark

### Question

**tl;dr**  I'm having issues writing Iceberg tables to path-based locations. What's the correct way to do it?
________

The [official documentation](https://iceberg.apache.org/docs/latest/spark-writes/#creating-tables) states that path-based writes are supported:

> The Iceberg table location can also be specified by the location table property:
> ```python
> data.writeTo("prod.db.table")
>     .tableProperty("location", "/path/to/location")
>     .createOrReplace()
> ```

I followed it and tried to create a table in a custom path in this way:

```python
df.writeTo('my_table') \
  .using('iceberg') \
  .tableProperty('location', '/home/iceberg/warehouse/test/custom/location/my_table') \
  .createOrReplace()
```

but got this error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/spark/python/pyspark/sql/readwriter.py", line 1453, in createOrReplace
    self._jwriter.createOrReplace()
  File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
  File "/opt/spark/python/pyspark/sql/utils.py", line 196, in deco
    raise converted from None
pyspark.sql.utils.IllegalArgumentException: Cannot set a custom location for a path-based table. Expected /home/iceberg/warehouse/default/my_table but got /home/iceberg/warehouse/test/custom/location/my_table
```

My Spark session is configured as follows:
```
spark.sql.extensions: org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions
spark.sql.catalog.spark_catalog: org.apache.iceberg.spark.SparkSessionCatalog
spark.sql.catalog.spark_catalog.type: hadoop
spark.sql.catalog.spark_catalog.warehouse: /home/iceberg/warehouse
spark.sql.defaultCatalog: spark_catalog
```
 
I also tried different configurations (e.g., adding another catalog with a `org.apache.iceberg.spark.SparkCatalog` implementation) without success.

Am I doing something wrong or is the documentation out-of-date?

Any support would be appreciated — thanks everyone!

**Comment by nastra** [comment]
Your warehouse is already path-based due to setting `spark.sql.catalog.spark_catalog.warehouse: /home/iceberg/warehouse` and so the `HadoopCatalog` is preventing the use of `location` for a table that is already path-based.
I believe this should work with any other catalog other than `hadoop`

**Comment by pietro-agilelab** [comment]
Thanks a lot @nastra for your explanation!

Is there any way to specify only the path, so not to create any entry in the catalog?
For example something like:

```python
df.write.format("iceberg").mode("overwrite").save("/path/to/my/table")
```

**Comment by sebastien-namy-cbp** [comment]
@pietro-agilelab 
Did you find a way to specify the s3 path of the table in iceberg format when it was created? 

I have the same need

**Comment by pietro-agilelab** [comment]
Hi @sebastien-namy-cbp, after running some tests I came to the following conclusion:

### Hadoop catalog
If you are using a catalog of type "hadoop", then you can't use the `tableProperty("location", "s3://...")` property of the DataFrameWriterV2 API, otherwise you'll get [an error](https://github.com/apache/iceberg/issues/8377#issue-1862786604).

Similarly, creating a table with `df.write.format("iceberg").mode("overwrite").save("/path/to/my/table")` raises [an error](https://github.com/apache/iceberg/issues/8377#issuecomment-1691790773) too.

The only workaround I found is by using the DataFrameWriterV2 API: assuming that you configured an Iceberg catalog called "local" with a warehouse path `s3://my-bucket/warehouse`, then writing
```python
df.writeTo("local.db.my.table").using("iceberg").createOrReplace()
```
will create a table at location `s3://my-bucket/warehouse/local/db/my/table`.


### Hive catalog
If instead you are using a catalog of type "hive", then you can follow the example of the [documentation](https://iceberg.apache.org/docs/latest/spark-writes/#creating-tables), thus writing:
```python
# creates table `iceberg.db.table` at location `s3://my-bucket/path/to/location`
df.writeTo("iceberg.db.table")
    .tableProperty("location", "s3://my-bucket/path/to/location")
    .createOrReplace()
```

**Comment by sebastien-namy-cbp** [comment]
Thank you for your answer
